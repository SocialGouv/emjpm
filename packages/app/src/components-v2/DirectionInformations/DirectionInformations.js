import { Card, Heading3, Heading5 } from "@socialgouv/emjpm-ui-core";
import React from "react";
import { Box, Flex, Text } from "rebass";
import { MailOutline } from "styled-icons/material";

import { LinkButton } from "../Commons";
import { boxStyle, flexStyle, iconTextStyle, innerTextStyle } from "./style";

const DirectionInformations = props => {
  const { email, nom, prenom } = props;
  return (
    <Box {...props}>
      <Card p="5">
        <Heading3>
          {nom ? nom : "Nom non renseigné"} {prenom ? prenom : "Prénom non renseigné"}
        </Heading3>
        <Flex sx={flexStyle}>
          <Box sx={boxStyle}>
            <Heading5>Vos informations</Heading5>
            <Flex mt="2">
              <MailOutline size="16" />
              <Text sx={iconTextStyle}>{email}</Text>
            </Flex>
            <Text sx={innerTextStyle}>{nom ? nom : "Nom non renseigné"} </Text>
            <Text sx={innerTextStyle}>{prenom ? prenom : "Prénom non renseigné"}</Text>
          </Box>
        </Flex>
        <Flex mt="5">
          <Box>
            <LinkButton href="/direction/edit-informations">Modifier vos informations</LinkButton>
          </Box>
        </Flex>
        <Flex mt="1">
          <LinkButton href="/direction/edit-password">Modifier votre mot de passe</LinkButton>
        </Flex>
      </Card>
    </Box>
  );
};

export { DirectionInformations };
