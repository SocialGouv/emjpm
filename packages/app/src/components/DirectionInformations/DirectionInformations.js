import { directionFormatter } from "@emjpm/core";
import { Card, Heading3, Heading5 } from "@emjpm/ui";
import { Building } from "@styled-icons/fa-regular/Building";
import { MailOutline } from "@styled-icons/material/MailOutline";
import { PersonOutline } from "@styled-icons/material/PersonOutline";
import React, { useContext } from "react";
import { Box, Flex, Text } from "rebass";

import { LinkButton } from "../Commons";
import { UserContext } from "../UserContext";
import { boxStyle, flexStyle, iconTextStyle } from "./style";

const DirectionInformations = (props) => {
  const { email, nom, prenom, directions } = useContext(UserContext);
  return (
    <Box {...props}>
      <Card p="5">
        <Heading3>
          {nom ? nom : ""} {prenom ? prenom : ""}
        </Heading3>
        <Flex sx={flexStyle}>
          <Box sx={boxStyle}>
            <Heading5>Vos informations</Heading5>
            <Flex mt="2">
              <Building size="16" />
              <Text sx={iconTextStyle}>
                {directionFormatter.formatAdministrationInfo(directions[0])}
              </Text>
            </Flex>
            <Flex mt="2">
              <MailOutline size="16" />
              <Text sx={iconTextStyle}>{email}</Text>
            </Flex>
            <Flex mt="2">
              <PersonOutline size="20" />
              <Text sx={iconTextStyle}>{`${prenom ? prenom : ""} ${nom ? nom : ""} `} </Text>
            </Flex>
          </Box>
        </Flex>
        <Flex mt="5">
          <Box>
            <LinkButton href="/direction/edit-informations">Modifier vos informations</LinkButton>
          </Box>
        </Flex>
        <Flex mt="1">
          <LinkButton href="/direction/edit-password">Modifier votre mot de passe</LinkButton>
        </Flex>
      </Card>
    </Box>
  );
};

export { DirectionInformations };
