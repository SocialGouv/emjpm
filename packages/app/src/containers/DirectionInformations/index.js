import { DirectionInformations } from "./DirectionInformations";

export { DirectionInformations };
