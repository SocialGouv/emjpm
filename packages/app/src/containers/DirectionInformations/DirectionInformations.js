import { directionFormatter } from "@emjpm/biz";
import { Building } from "@styled-icons/fa-regular/Building";
import { MailOutline } from "@styled-icons/material/MailOutline";
import { PersonOutline } from "@styled-icons/material/PersonOutline";
import { Box, Flex, Text } from "rebass";

import { LinkButton } from "~/containers/Commons";
import useUser from "~/hooks/useUser";
import { Card, Heading } from "~/components";

import { boxStyle, flexStyle, iconTextStyle } from "./style";

function DirectionInformations(props) {
  const { email, nom, prenom, directions } = useUser();
  return (
    <Box {...props} id="informations_generales" tabIndex="-1">
      <Card p="5">
        <Heading size={3}>
          {nom ? nom : ""} {prenom ? prenom : ""}
        </Heading>
        <Flex sx={flexStyle}>
          <Box sx={boxStyle}>
            <Heading size={5}>Vos informations</Heading>
            <Flex mt="2">
              <Building size="16" />
              <Text sx={iconTextStyle}>
                {directionFormatter.formatAdministrationInfo(directions[0])}
              </Text>
            </Flex>
            <Flex mt="2">
              <MailOutline size="16" />
              <Text sx={iconTextStyle}>{email}</Text>
            </Flex>
            <Flex mt="2">
              <PersonOutline size="20" />
              <Text sx={iconTextStyle}>
                {`${prenom ? prenom : ""} ${nom ? nom : ""} `}{" "}
              </Text>
            </Flex>
          </Box>
        </Flex>
        <Flex mt="5">
          <Box>
            <LinkButton
              to="/direction/edit-informations"
              title="Modifier vos informations"
              aria-label="Modifier vos informations"
            >
              Modifier vos informations
            </LinkButton>
          </Box>
        </Flex>
        <Flex mt="1">
          <LinkButton
            to="/direction/edit-password"
            title="Modifier votre mot de passe"
            aria-label="Modifier votre mot de passe"
          >
            Modifier votre mot de passe
          </LinkButton>
        </Flex>
      </Card>
    </Box>
  );
}

export { DirectionInformations };
