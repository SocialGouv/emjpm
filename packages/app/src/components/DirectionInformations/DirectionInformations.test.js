import React from "react";
import renderer from "react-test-renderer";

import { DirectionInformations } from "./DirectionInformations";

it("renders correctly", () => {
  const tree = renderer.create(<DirectionInformations />).toJSON();
  expect(tree).toMatchSnapshot();
});
