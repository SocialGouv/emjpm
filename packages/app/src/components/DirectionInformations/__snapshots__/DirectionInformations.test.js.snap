// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-vurnku"
>
  <div
    className="css-c0gbp4"
  >
    <div
      className="css-yqshnb"
    >
      Nom non renseigné
       
      Prénom non renseigné
    </div>
    <div
      className="css-3inb56"
    >
      <div
        className="css-wcquo4"
      >
        <div
          className="css-1htq42o"
        >
          Vos informations
        </div>
        <div
          className="css-1i0hao9"
        >
          <svg
            aria-hidden="true"
            className="sc-bwzfXH bqgLxe"
            fill="currentColor"
            focusable="false"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
            />
          </svg>
          <div
            className="css-z0hm0k"
          />
        </div>
        <div
          className="css-fuipfr"
        >
          Nom non renseigné
           
        </div>
        <div
          className="css-fuipfr"
        >
          Prénom non renseigné
        </div>
      </div>
    </div>
    <div
      className="css-c6ndte"
    >
      <div
        className="css-vurnku"
      >
        <a
          className="css-5xn953"
          href="/direction/edit-informations"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Modifier vos informations
        </a>
      </div>
    </div>
    <div
      className="css-arxxmz"
    >
      <a
        className="css-5xn953"
        href="/direction/edit-password"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Modifier votre mot de passe
      </a>
    </div>
  </div>
</div>
`;
